server/plugin/api_test.go

```diff
@@ -1112,3 +1197,102 @@ func TestHandleGetGitRepositoryBranches(t *testing.T) {
+	for _, testCase := range []struct {
+		description                            string
+		body                                   string
+		getSubscriptionFilterPossibleValuesErr error
+		statusCode                             int
+		getGitRepositoryBranchesResponse       *serializers.SubscriptionFilterPossibleValuesResponseFromClient
+		expectedResponse                       string
+		expectedErrorResponse                  interface{}
+	}{
+		{
+			description: "HandleGetSubscriptionFilterPossibleValues: valid",
+			body: `{
+				"organization": "mockOrganization",
+				"projectId": "mockProjectID",
+				"eventType": "mockEventType",
+				"repositoryId": "mockRepositoryID",
+				"filters": ["mockFilter1", "mockFilter2"]
+				}`,
+			statusCode: http.StatusOK,
+			getGitRepositoryBranchesResponse: &serializers.SubscriptionFilterPossibleValuesResponseFromClient{
+				InputValues: []*serializers.InputValues{
+					{
+						PossibleValues: []*serializers.PossibleValues{},
+						SubscriptionFilter: serializers.SubscriptionFilter{
+							InputID: "mockInputID1",
+						},
+					},
+					{
+						PossibleValues: []*serializers.PossibleValues{},
+						SubscriptionFilter: serializers.SubscriptionFilter{
+							InputID: "mockInputID2",
+						},
+					},
+				},
+			},
+			expectedResponse: `{"mockInputID1":[],"mockInputID2":[]}`,
+		},
+		{
+			description: "HandleGetSubscriptionFilterPossibleValues: missing fields",
+			body: `{
+				"projectId": "mockProjectID",
+				"eventType": "mockEventType",
+				"repositoryId": "mockRepositoryID",
+				"filters": ["mockFilter1", "mockFilter2"]
+				}`,
+			statusCode:            http.StatusBadRequest,
+			expectedErrorResponse: map[string]interface{}{"Error": constants.OrganizationRequired},
+		},
+		{
+			description: "HandleGetSubscriptionFilterPossibleValues: Error fetching subscription filter possible values",
+			body: `{
+				"organization": "mockOrganization",
+				"projectId": "mockProjectID",
+				"eventType": "mockEventType",
+				"repositoryId": "mockRepositoryID",
+				"filters": ["mockFilter1", "mockFilter2"]
+				}`,
+			statusCode:                             http.StatusInternalServerError,
+			getSubscriptionFilterPossibleValuesErr: errors.New("failed to fetch subscription filters possible values"),
+			expectedErrorResponse:                  map[string]interface{}{"Error": "failed to fetch subscription filters possible values"},
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			mockAPI.On("LogError", testutils.GetMockArgumentsWithType("string", 3)...)
+
+			if testCase.statusCode == http.StatusOK || testCase.statusCode == http.StatusInternalServerError {
+				mockedClient.EXPECT().GetSubscriptionFilterPossibleValues(gomock.Any(), gomock.Any()).Return(testCase.getGitRepositoryBranchesResponse, testCase.statusCode, testCase.getSubscriptionFilterPossibleValuesErr)
+			}
+
+			req := httptest.NewRequest(http.MethodGet, "/mockPath", bytes.NewBufferString(testCase.body))
```

Comment:
not able to understand this you are not calling the actual API and what is the need for this then ?

Reply:
We are adding the `Mattermost-User-ID` header in this req and passing it to the function call